coalib/parsing/CliParsing.py

```diff
@@ -87,3 +87,31 @@ def parse_custom_settings(sections,
                                origin=origin,
                                section_name=key_touple[0],
                                from_cli=True)
+
+
+def check_conflicts(cli_sections, log_printer):
+    '''
+    Checks if there are any conflicting aruments passed
+
+    :return:            True if no conflicts
+    :raises SystemExit: If there are conflicting arguments (exit code: 2)
+    '''
+    conflicts = defaultdict(set)
+
+    # Add any conflict definitions here
+    conflicts["no_config"].add("save")
+    conflicts["no_config"].add("find_config")
+
+    args = list(conflicts.keys())
+    for arg in args:
+        for conflict in conflicts[arg]:
+            conflicts[conflict].add(arg)
+
+    for section in cli_sections:
+        for arg in cli_sections[section]:
+            for conflict in conflicts[arg]:
```

Comment:
In general if you ahve a for followed by a single if inside a set is
probably what you want with the & operator. As that's exactly what that
does i think
Ive never profiled it myself though

On Sun, Mar 13, 2016 at 11:37 AM, Adhityaa Chandrasekar <
notifications@github.com> wrote:

> In coalib/parsing/CliParsing.py
> <https://github.com/coala-analyzer/coala/pull/1863#discussion_r55932670>:
>
> > +    :raises SystemExit: If there are conflicting arguments (exit code: 2)
> > +    '''
> > +    conflicts = defaultdict(set)
> > +
> > +    # Add any conflict definitions here
> > +    conflicts["no_config"].add("save")
> > +    conflicts["no_config"].add("find_config")
> > +
> > +    args = list(conflicts.keys())
> > +    for arg in args:
> > +        for conflict in conflicts[arg]:
> > +            conflicts[conflict].add(arg)
> > +
> > +    for section in cli_sections:
> > +        for arg in cli_sections[section]:
> > +            for conflict in conflicts[arg]:
>
> I was talking with @sils1297 <https://github.com/sils1297> yesterday. Is
> this http://ix.io/rrS/py better? It's simpler and removes the redundant
> lines [105-109].
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/coala-analyzer/coala/pull/1863/files#r55932670>.
>